File: sample/sort/anagram-sort/anagram-sort.py

a = ["cb", "abc", "a", "bb", "bac", "ca", "d", "cba", "bc", "acb"]


def cmp(x, y):
    if x < y:
        return -1
    elif x > y:
        return 1
    else:
        return 0

print(sorted(a, cmp=cmp))

File: sample/sort/anagram-sort/anagram-sort.hs
import Data.List

-- MEMO: :Q:anagram同士が隣同士になるようソート(CI9.2)
sort' :: [String] -> [String]
sort' = Data.List.sortBy (\a b -> compare (sort a) (sort b))

main = do
  print $ sort' ["cb", "abc", "a", "bb", "bac", "ca", "d", "cba", "bc", "acb"]
